src/core/email-client.js

```diff
@@ -53,101 +71,225 @@ function getFreshImap (cb) {
+    }
+
+    _loadChangeCheck(){
         var self = this;
+        return self._changeCheck.promise;
+    }
+
+    //method to change name of user
+    _changeName(name){
+        var self = this;
+        self._name = name;
+    }
+
+    _changeEmail(email, password){
+        var self = this;
+        //self._email = email;
+
+        self._imap = Q.defer();
+        self._transporter = Q.defer();
+
+        var obj = {
+          "email": email,
+          "password": password
+        };
 
-        self.imap.on('ready', function () {
-            self.imap.openBox('INBOX', false, function (err, box) {
+        jsonfile.writeFile("./email.json", obj, function (err) {
+            if (err){
+                console.log(err);
+                return;
+            }
+            console.log("email.json was updated!");
+
+            getFreshImap(function (err, imap, email, smtpConfig) {
                 if (err) {
-                    throw err;
+                    console.log(err);
+                } else {
+                    self._imap.resolve(imap);
+                    self._email = email;
+                    self._changeCheck.resolve(true);
+                    self._transporter.resolve(NodeMailer.createTransport(smtpConfig));
+                    
                 }
+            });            
 
-                //  Parses all unseen emails and marks them as read
-                var collectUnseenEmails = function () {
-                    self.imap.search(['UNSEEN'], function (err, results) {
-                        if (err || results.length === 0) {
-                            console.log(err || 'NO NEW MAIL');
-                            return;
-                        }
-
-                        //  Fetch the entire email bodies
-                        var f = self.imap.fetch(results, {
-                            bodies: [''],
-                            markSeen: true
-                        });
+        });
 
-                        f.on('message', function (msg, seqno) {
-                            var parser = new MailParser();
-                            parser.on('end', function (mail) {
-                                //  Done parsing
-                                //  TODO: logic for sending event to event bus goes here
-                                console.log('NEW MAIL!');
-                                console.log(
-                                    'From: ' + mail.from[0].address + '\n' +
-                                    'Subject: ' + mail.subject + '\n' +
-                                    'Body:\n' + mail.text
-                                );
+
+    }
+
+    //attachments as array of Path
+    _sendEmail(toName, toEmail, subject, text, attachments){
+        var self = this;
+        var fromText =  '"' + self.name + '" <' + self._email + '>'; 
+        var toText = '"' + toName + '" <' + toEmail + '>'; 
+
+        var files = [];
+        if (attachments != null){
```

Comment:
Use strict compare: !==
Lint should've caught this actually, can you verify this for me?